core/sdk-core/src/main/java/software/amazon/awssdk/core/progress/snapshot/ListenerProgress.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.progress.snapshot;
+
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public interface ListenerProgress {
```

Comment:
What's this used for?

Reply:
Transfer Listener uses this as a wrapping class around Snapshot; it ideally defines methods to update (or create copies and update) the snapshot associated with it, like https://github.com/aws/aws-sdk-java-v2/blob/0b9b0735c447af78316e5d7ef960e2b237f2f073/services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/progress/DefaultTransferProgress.java#L38